### PR TITLE
wip: feat(component): Create PostCard and LastTenPosts component

### PR DESCRIPTION
**Pull request by agory**
# What is the purpose of your pull request ?

- [] new blog post.
- [] bug fix
- [X] new feature for Open Blog

# Describe your PR content

Add LastTenArticles component. It displays the last 10 articles of a specific language.

### CODE REVIEW

File: docs/.vuepress/components/LastTenArticles.vue [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  }
+}
+
+export default {
+  name: "LastTenArticles",
+  props: ["locale"],
+  computed: {
+    lastTenArticles() {
+      return this.$site.pages
+        .map(page => new Article(page))
```
Comment (Slashgear) [inline docs/.vuepress/components/LastTenArticles.vue]: Could we filter on type === post ?

File: docs/.vuepress/components/LastTenArticles.vue [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        .filter(filterByLocale(this.locale, this.$site.locales))
```
Comment (Slashgear) [inline docs/.vuepress/components/LastTenArticles.vue]: We may use https://v1.vuepress.vuejs.org/plugin/option-api.html#extendpagedata to add local key in `$page`

File: docs/README.md [hunk 3]
```diff
@@ -10,3 +10,5 @@ features:
     details: Thanks to VuePress, you could add/translate freely content in your local.
 footer: MIT Licensed | Copyright © 2019-present Antoine Caron
 ---
+
+<LastTenArticles locale="en-US"/>
```
Comment (Slashgear) [inline docs/README.md]: With local in `$page` this props should ne be required then 😉

File: docs/.vuepress/components/LastTenArticles.vue (hunk 1 above)
Comment (Slashgear) [inline docs/.vuepress/components/LastTenArticles.vue]: Creating Type Article is a bit too much  here. We could just return an `{}` object

**Review by Slashgear — COMMENTED** [review]
Very nice 🆗

**Comment by Slashgear** [comment]
Integration is broken:

- [ ] responsive is broken, maybe we should display only one item by row 
- [ ] There is no gap between "card" which make it hard to read.
Capture d’écran 2019-04-07 à 19 37 28

- [ ] Local Filter is not working 
- [ ] Could we add this component at the end of articles ? 
- [ ] Colors should be defined in `palette.styl`
- [ ] Could we use `object-fit` on images ?

Don't hesitate if you need help ;)

File: docs/.vuepress/components/ArticleCard.vue [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+.card-title {
+  padding-bottom: 0.5em;
+  padding-left: 0.5em;
+  font-size: 1.2em;
+  height: 3.6em;
+  line-height: 1.2em;
+  overflow: hidden;
+}
+a {
```
Comment (Slashgear) [inline docs/.vuepress/components/ArticleCard.vue]: Could we remove this `a` override ?

File: docs/.vuepress/components/LastTenArticles.vue (hunk 1 above)
Comment (agory) [reply docs/.vuepress/components/LastTenArticles.vue]: done

File: docs/.vuepress/components/LastTenArticles.vue (hunk 2 above)
Comment (agory) [reply docs/.vuepress/components/LastTenArticles.vue]: done

File: docs/README.md (hunk 3 above)
Comment (agory) [reply docs/README.md]: done

File: docs/.vuepress/components/LastTenArticles.vue (hunk 1 above)
Comment (agory) [reply docs/.vuepress/components/LastTenArticles.vue]: done

File: docs/.vuepress/components/ArticleCard.vue (hunk 4 above)
Comment (agory) [reply docs/.vuepress/components/ArticleCard.vue]: done

**Comment by agory** [comment]
> Integration is broken:
> 
> * [X]  responsive is broken, maybe we should display only one item by row
> * [X]  There is no gap between "card" which make it hard to read.
> 
> Capture d’écran 2019-04-07 à 19 37 28
> 
> * [X]  Local Filter is not working
> * [X]  Could we add this component at the end of articles ?
> * [X]  Colors should be defined in `palette.styl`
> * [X]  Could we use `object-fit` on images ?
> 
> Don't hesitate if you need help ;)

File: docs/.vuepress/components/PostCard.vue [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+.card-author-name
+  padding-left: 1em
+
+
+.card-title
+  padding-bottom: 0.5em
+  padding-left: 0.5em
+  font-size: 1.2em
+  height: 3.6em
+  line-height: 1.2em
```
Comment (Slashgear) [inline docs/.vuepress/components/PostCard.vue]:
```suggestion
  line-height: 1.4;
```

To fix title when it is too long

**Review by Slashgear — APPROVED** [review]

**Pull request merged**